ui/app/AppLayouts/Chat/ContactsColumn.qml

```diff
@@ -10,23 +10,233 @@ import "./components"
+
+            StatusBaseText {
+                font.pixelSize: 15
+                color: Theme.palette.directColor5
+                anchors.centerIn: parent
+                text: qsTr("No search results")
+            }
+        }
+
+        StatusChatList {
+            id: channelList
+
+            chatNameFn: function (chatItem) {
+                return chatItem.chatType !== Constants.chatTypePublic ?
+                    Emoji.parse(Utils.removeStatusEns(Utils.filterXSS(chatItem.name))) :
+                    Utils.filterXSS(chatItem.name)
+            }
+
+            profileImageFn: function (id) {
+                return appMain.getProfileImage(id)
+            }
+
+            filterFn: function (chatListItem) {
+                return !!!contactsColumn.searchStr || chatListItem.name.toLowerCase().includes(contactsColumn.searchStr.toLowerCase())
+            }
+
+            Connections {
+                target: profileModel.contacts.list
+                onContactChanged: {
+                    for (var i = 0; i < channelList.chatListItems.count; i++) {
+                        let chatItem = channelList.chatListItems.itemAt(i);
+                        if (chatItem.chatId === pubkey) {
+                            let profileImage = appMain.getProfileImage(pubkey)
+                            if (!!profileImage) {
+                                chatItem.image.isIdenticon = false
+                                chatItem.image.source = profileImage
+                            }
+                            break;
+                        }
+                    }
+                }
+            }
+
+            chatListItems.model: chatsModel.channelView.chats
+            selectedChatId: chatsModel.channelView.activeChannel.id
+
+            onChatItemSelected: chatsModel.channelView.setActiveChannel(id)
+            onChatItemUnmuted: chatsModel.channelView.unmuteChatItem(id)
+
+            popupMenu: StatusPopupMenu {
+
+                id: chatListContextMenu
+
+                property var chatItem
+
+                openHandler: function (id) {
+                    chatItem = chatsModel.channelView.getChatItemById(id)
+                }
+
+                StatusMenuItem {
+                    id: viewProfileMenuItem
+                    text: {
+                        if (chatItem) {
+                            switch (chatItem.chatType) {
+                                case Constants.chatTypeOneToOne:
+                                  return qsTr("View Profile")
+                                  break;
+                                case Constants.chatTypePrivateGroupChat:
+                                  return qsTr("View Group")
+                                  break;
+                                default:
+                                  return qsTr("Share Chat")
+                                  break;
+                            }
+                        }
+                        return ""
```

Comment:
This switch case construct looks weird. The only reason it exists is because QML seems to parse the menu item's `text` property, even though the popup menu hasn't been opened yet (at which point `chatItem` is undefined, which results in runtime errors. Luckily they don't break anything, but it still doesn't look nice to be bombarded with unnecessary errors).